src/hooks/index.js

```diff
@@ -0,0 +1,97 @@
+import api from "../api/api";
+import { Server } from "../utils/config";
+import { useEffect, useReducer } from "react";
+
+export const FetchState = {
+  FETCH_INIT: 0,
+  FETCH_SUCCESS: 1,
+  FETCH_FAILURE: 2,
+};
+
+export const useGetTodos = (stale) => {
+  const reducer = (state, action) => {
+    switch (action.type) {
+      case FetchState.FETCH_INIT:
+        return { ...state, isLoading: true, isError: false };
+      case FetchState.FETCH_SUCCESS:
+        return {
+          ...state,
+          isLoading: false,
+          isError: false,
+          todos: action.payload,
+        };
+      case FetchState.FETCH_FAILURE:
+        return { ...state, isLoading: false, isError: true };
+      default:
+        throw new Error();
+    }
+  };
+
+  const [state, dispatch] = useReducer(reducer, {
+    isLoading: false,
+    isError: false,
+    todos: [],
+  });
+
+  useEffect(() => {
+    let didCancel = false;
+    const getTodos = async () => {
+      dispatch({ type: FetchState.FETCH_INIT });
+      try {
+        const data = await api.listDocuments(Server.collectionID);
+        if (!didCancel)
```

Comment:
This might fail when using a lint. use a curly braces here.

Reply:
okay 👍

Reply:
okay 👍

Reply:
I ran an eslint but it did not throw any errors 🤔